pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -493,32 +493,10 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 		uniqueVolumeName = util.GetUniqueVolumeNameFromSpecWithPod(volume.podName, plugin, volumeSpec)
 	}
 
-	volumeMounter, newMounterErr := plugin.NewMounter(
-		volumeSpec,
-		pod,
-		volumepkg.VolumeOptions{})
-	if newMounterErr != nil {
-		return nil, fmt.Errorf(
-			"reconstructVolume.NewMounter failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v",
-			uniqueVolumeName,
-			volumeSpec.Name(),
-			volume.podName,
-			pod.UID,
-			newMounterErr)
-	}
-
-	// Check existence of mount point for filesystem volume or symbolic link for block volume
-	isExist, checkErr := rc.operationExecutor.CheckVolumeExistenceOperation(volumeSpec, volumeMounter.GetPath(), volumeSpec.Name(), rc.mounter, uniqueVolumeName, volume.podName, pod.UID, attachablePlugin)
-	if checkErr != nil {
-		return nil, checkErr
-	}
-	// If mount or symlink doesn't exist, volume reconstruction should be failed
-	if !isExist {
-		return nil, fmt.Errorf("Volume: %q is not mounted", uniqueVolumeName)
-	}
+	var volumeMapper volumepkg.BlockVolumeMapper
+	var volumeMounter volumepkg.Mounter
 
 	// TODO: remove feature gate check after no longer needed
-	var volumeMapper volumepkg.BlockVolumeMapper
 	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) && volume.volumeMode == v1.PersistentVolumeBlock {
 		var newMapperErr error
 		if mapperPlugin != nil {
```

Comment:
It is not your change, but I don't see the possibility of mapperPlugin is nil?

Reply:
@jingxu97 

Thank you for your comment.

You're right that this path won't be reached, beause [`ReconstructVolumeOperation` returns error when `mapperPlugin == nil`](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L901) just after [`FindMapperPluginByName` returns `nil` for `mapperPlugin`](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L1022) when plugin doesn't support block volume.

However, it doesn't seem to be so obvious, because it depends on other function's implementation. 
In this case, do you mean to delete this check here or any other ideas for improve this?

Reply:
It just seems strange to check this quite late here.

Reply:
@jingxu97 @oomichi 

I've moved nil check for `mapperPlugin` earlier in `reconstructVolume` in "Move nil check for mapperPlugin earlier in reconstructVolume" commit. PTAL

(I will merge this change with the previous commit if this looks good.)